stl/inc/xstring

```diff
@@ -3371,11 +3371,9 @@ public:
         _In_reads_(_Count) const _Elem* const _Ptr, _CRT_GUARDOVERFLOW const size_type _Count) {
         // assign [_Ptr, _Ptr + _Count)
         if (_Count <= _Mypair._Myval2._Myres) {
-            _ASAN_STRING_MODIFY(*this, _Mypair._Myval2._Mysize, _Count);

```

Comment:
I'm afraid that it won't be OK to just remove `_ASAN_STRING_MODIFY` here. Should we move it immediately before `return *this`?

Reply:
`_Eos` does `_ASAN_STRING_MODIFY`

Reply:
If the argument doesn't alias `*this`, we could be growing the string, so we need to "unpoison" the extra size. If the argument _does_ alias `*this` and we're shrinking, we need to wait until after copying the characters to poison the removed size. Altogether, it's best to just remove and recreate here.